gazel/gazel.go

```diff
@@ -49,9 +53,17 @@ func main() {
 	if err := v.walkRepo(); err != nil {
 		glog.Fatalf("err walking repo: %v", err)
 	}
-	if _, err := v.reconcileAllRules(); err != nil {
+	written := 0
+	if written, err = v.reconcileAllRules(); err != nil {
 		glog.Fatalf("err reconciling rules: %v", err)
 	}
+	if *validate {
+		if written > 0 {
+			fmt.Fprintf(os.Stderr, "\n%d BUILD files not up-to-date.\n", written)
+			os.Exit(1)
+		}
+		fmt.Fprintf(os.Stderr, "All BUILD files up-to-date.\n")
```

Comment:
I don't have a source, but I'm under the impression that if a validation-type-thing succeeds, the Unix Way is to exit 0 and print nothing. gofmt and go vet follow this, for instance.